src/modm/platform/timer/stm32/general_purpose_base.hpp.in

```diff
@@ -275,6 +275,16 @@ protected:
 	signalToChannel()
 	{
 		modm::platform::detail::SignalConnection<Signal, p>{};
+%% if target.family == "g4"
```

Comment:
Are you sure? There was another PR to add this, and I thought the problem was exactly that some of the f-series had different timer setups.

Reply:
fixed the STM-family ifs in three places

Reply:
So now there are the following incompatibilities for the signalToChannel:
1. f101 and f102 don't have ChXn channels, but f103 has because of different timer sets
2. for f410tby the ChXn are not generated, as there are no corresponding pins, but the timers support the feature. 

What would be the best way to resolve this?

Reply:
I have pushed a commit that obtains the available signals from the device data. Let's see if that works.

> for f410tby the ChXn are not generated, as there are no corresponding pins, but the timers support the feature.

Is that a big issue? The signal API is normally used with Gpio pins, like in `GpioA3::Ch3`. The APIs taking a numeric index would still work. Unfortunately, there doesn't seem to be an easily usable machine readable data source for timer channel information. We would have to hard code that for all device families.

Reply:
Regarding f410tby: the pins don't have the Chxn channels, so gpio::Signal enum doesn't have them either, and we have to avoid generation of these lines in this case. But I don't know how to detect this

Reply:
Your commit should solve this. It looks much nicer, than before

Reply:
With the change I have pushed only the if-branches for signals that exist in the gpio data are generated. In `module.lb` the signal data is now queried from the gpio module and only the cases for which pins exist are generated.